Add TaskConnectionHandler and TaskHttpConnectionHandler

david-eckardt-sociomantic opened the pull request:
`TaskConnectionHandler` and `TaskHttpConnectionHandler` are the task based equivalents to `IFiberConnectionHandler` and `HttpConnectionHandler`.

david-eckardt-sociomantic committed:
Add TaskConnectionHandler and TaskHttpConnectionHandler

gavin-norman-sociomantic commented on src/ocean/net/server/connection/TaskConnectionHandler.d:
```diff
@@ -0,0 +1,166 @@
+
+/*******************************************************************************
+
+    Connection handler abstract base class.
```
If you removed the `/* Imports */` comment, you could use `/// ditto` here.

gavin-norman-sociomantic commented on test/httpserver/main.d:
```diff
@@ -0,0 +1,249 @@
+    functions, etc) which could, under certain environmental conditions, fail.
+
+    Copyright:      Copyright (c) 2017 sociomantic labs. All rights reserved
+
```
License missing.

gavin-norman-sociomantic commented on src/ocean/net/http/TaskHttpConnectionHandler.d:
```diff
@@ -0,0 +1,440 @@
+
+/******************************************************************************/
+
+abstract class TaskHttpConnectionHandler : TaskConnectionHandler
```
Same comment about `/* Imports */` and `/// ditto`.

gavin-norman-sociomantic commented on src/ocean/net/http/TaskHttpConnectionHandler.d:
```diff
@@ -0,0 +1,440 @@
+
+    /**************************************************************************
+
+        HTTP request message parser and response message generator
```
A small point, but I don't like "X and Y" comments like this. It's clearer to have a separate comment for each member.

gavin-norman-sociomantic commented on src/ocean/net/http/TaskHttpConnectionHandler.d:
```diff
@@ -0,0 +1,440 @@
+
+     **************************************************************************/
+
+    protected HttpException             http_exception;
```
Apparently no purpose for the extra indentation.

gavin-norman-sociomantic replied:
(Not reviewing the rest of the code in this module as I presume it's just copied from the existing HTTP connection handler.)

mihails-strasuns-sociomantic commented:
Ping @david-eckardt-sociomantic :)

david-eckardt-sociomantic commented:
Updated, addressing all comments and adding a warning to the release notes that `Task(Http)ConnectionHandler` hasn't been tested for performance in a production environment yet.

mihails-strasuns-sociomantic commented:
OK, let's do it. I will probably have more detailed look myself next time I get to working on task features and optimization as it looks like great source of practical cases.
